tests/main.cpp

```diff
@@ -16,7 +16,7 @@ int main(int argc, char* argv[])
 	int failed = 0;
 	for (QTestSuite*& suite : QTestSuite::suites())
 	{
-		failed += QTest::qExec(suite, argc, argv);
+		if (QTest::qExec(suite, argc, argv) != 0) { ++failed; }
```

Comment:
Probably out of scope?

Reply:
I can see that point of view, and can revert it if you want. However, I'm not sure it warrants its own PR. and this PR does at least touch the tests, so it's not entirely irrelevant.

I don't think I've mentioned the reason for the change, so I'll explain it here. The [documentation for `QTest::qExec`](https://doc.qt.io/qt-5/qtest.html#qExec) states "this function returns 0 if no tests failed, or a value other than 0 if one or more tests failed or in case of unhandled exceptions". As such, this is a more appropriate way to use it than the original, where the specific non-zero value was relevant. In practise, it seems to return the number of tests in the suite which failed, which is not useful when trying to count the number of failed suites, not tests. I made this fix when I got a nonsensical output while developing the tests included in this PR.

Reply:
> In practise, it seems to return the number of tests in the suite which failed, which is not useful when trying to count the number of failed suites, not tests.

I do agree. It seems more realistic to count the number of suites that failed rather than the number of failed tests in each suite. Also, the documentation stating "this function returns 0 if no tests failed, or a value other than 0 if one or more tests failed or in case of unhandled exceptions" doesn't really guarantee that it will return a specific kind of number on failure (i.e., # of failed tests), just non-zero, and that should be the only thing we depend on. Feel free to keep the change, I don't really mind :+1:.